File: shortestToChar.js
function shortestToChar(S , C) {
    let index = [];
    let res = []
    for(let i =0 ; i< S.length;++i) {
        if( S[i] == C) index.push(i)
    }
    for(let j=0;j<S.length;++j){
        let i = 0;
        let minDistance = Infinity;
        while( i< index.length){
            if(Math.abs(index[i] - j) < minDistance) minDistance = Math.abs(index[i] - j);
            i++
        }
        res.push(minDistance)
    }
    return res
}
l

File: shortestToChar.py
class Solution:
    def shortestToChar(S , C) :
        index = [];
        res = [];
        for i in range(len(S)):
            if S[i] == C:
                index.append(i)
        for j in range(len(S)):
            i = 0 ;
            minDistance = len(S)+1
            while i < len(index):
                if abs(index[i]-j) < minDistance:
                    minDistance = abs(index[i]-j)
                i=i+1
            res.append(minDistance)
        return res
                 